### PR TITLE
Refactor main arguments + other improvements

### PR DESCRIPTION
**Pull request by simahawk**
Introduce 3 concepts:

* source: source of the comparison
* target: target of the comparison
* destination: where to push the work done

All of them accept full info as Github short entities like: org/$repo#$branch.

This is also a preliminary step to support porting from one repo to another (very handy in case of modules moved to different repos).

TODO

- [ ] when source and target are given only as branches it MUST use local branches
- [x] improve "No remote..." msg to show which ref is wrong
- [ ] print source, target, destination before running
- [ ] when destination is given create the local branch if not present, fetch before in case is pushed already
- [ ] when destination is given without branch (e.g. `--destination camptocamp/wms`), create one locally
- [ ] branch diff comparison: if destination is not given, always compare between source and target. If destination is given, check if destination exists (fetch it if a remote is given for it), if yes replace "target" in the comparison
- [ ] ...

### CODE REVIEW

File: oca_port/cli/main.py [hunk 1]
```diff
@@ -15,10 +15,17 @@
     $ export GITHUB_TOKEN=<token>
     $ oca-port 13.0 14.0 shopfloor --verbose
 
-To effectively migrate the addon or port its commits, use the `--fork` option:
+To effectively migrate the addon or port its commits, use the `--destination` option:
 
-    $ oca-port 13.0 14.0 shopfloor --fork camptocamp
+    $ oca-port 13.0 14.0 shopfloor --destination camptocamp/wms#14-mig-shopfloor
```
Comment (sebalix) [inline oca_port/cli/main.py:20]: Is there a commonly accepted standard to write this? `git` itself sometimes use `{remote}:{branch}` syntax, there is also `@` used to mention a reference, like in `git+https://github.com/OCA/oca-port.git@refs/pull/26/head`.

Should we support several patterns?

- `{org}/{repo}#{branch}`
- `{org}/{repo}@{branch}`
- `{org}/{repo}:{branch}`

File: oca_port/app.py [hunk 2]
```diff
@@ -19,24 +18,18 @@ class App(Output):
...
-        from_branch:
-            the source branch (e.g. '15.0')
-        to_branch:
-            the source branch (e.g. '16.0')
+        source:
+            SmartDict with branch, org, remote, source of the comparison
+        target:
+            SmartDict with branch, org, remote, target of the comparison
+        destination:
+            SmartDict with branch, org, remote, destination of the fork
```
Comment (sebalix) [inline oca_port/app.py:26]: Hum I'm not a fan of depending on `make_gh_info/SmartDict` internals to initialize an `App` object when using oca-port as a Python package, I find it complicated from a user POV, especially for mandatory parameters.

File: oca_port/cli/main.py [hunk 3]
```diff
@@ -112,20 +128,24 @@ def main(
             The PRs are found from FROM_BRANCH commits that do not exist in TO_BRANCH.
     The user will be asked if he wants to port them.
 
-        To start the migration process, the `--fork` option must be provided in
+        To start the migration process, the `--destination` option must be provided in
     order to push the resulting branch on the user's remote.
     """
+    source = make_gh_info("source", source)
+    target = make_gh_info("target", target)
+    destination = make_gh_info("destination", destination)
```
Comment (sebalix) [inline oca_port/cli/main.py:136]: I would move this in `App.__init__`, it's OK IMO to still parse `source` here to retrieve the repo, it'll be parsed a second time in `App` but that's something we can afford.

File: tests/common.py [hunk 4]
```diff
@@ -7,24 +7,24 @@
...
 
-class CommonCase(unittest.TestCase):
 
-    _settings = {
-        "branch1": "15.0",
-        "branch2": "16.0",
-        "branch3": "17.0",
-        "addon": "my_module",
-        "from_org": None,
-        "from_remote": None,  # We're testing locally without any remote
```
Comment (sebalix) [inline tests/common.py:19]: Not setting a remote in the tests was a quick-win to not trigger any call on GH API.
We'll have to mock GH API call if we set one.

Comment (simahawk) [reply tests/common.py:19]: I'll fix this but sooner or later we'll have to mock :)

**Comment by sebalix** [comment]
README has to be updated too

**Comment by sebalix** [comment]
I was thinking about another syntax to provide the organization and repository name alongside the branch, without re-inventing the wheel or the current syntax: `oca-port origin/15.0 another-repo/16.0 {ADDON} --destination fork/16.0-dev` and parse the remote URL of the different remotes to guess the organizations and repos to work on.

**Comment by simahawk** [comment]
continued and improved on https://github.com/OCA/oca-port/pull/51

**Pull request closed without merging**